<?xml version="1.0"?>

<!--
Fri Nov 11 14:26:18 EET 2016, Nikos Koukis

Launchfile for a simple demonstration of the mrpt_graphslam_2d algorithm on
single robot situations. More elaborate demo files are built on top of this one
- see sr_graphslam_demo_gt.launch file

-->


<launch>
    <arg name="disable_MRPT_visuals" default="false" />
	<arg name="robot_ns" value="/" />
    <arg name="output" value="screen" />
    <arg name="verbosity" value="1" />
    <arg name="bag_dir" value="$(find mrpt_graphslam_2d)/rosbags/demo_short_loop" />
    <arg name="bag_file" value="demo.bag"/>
    <arg name="bag_file_full" value="$(arg bag_dir)/$(arg bag_file)"/>
	<arg name="bag_topic_remaps" value="/scan:=input/laser_scan"/>
	<arg name="config_file" value="$(find mrpt_graphslam_2d)/config/ros_laser_odometry.ini"/>

    <!-- Should we also start an instance of rviz ? -->
    <arg name="start_rviz" value="true" />
    <arg name="rviz_file" value="$(find mrpt_graphslam_2d)/rviz/sr_graphslam.rviz" />

	<arg name="anchor_frame_ID" default="map" />
	<arg name="base_link_frame_ID" default="base_link"/>
	<arg name="laser_frame_ID" default="laser"/>
	<arg name="odometry_frame_ID" default="odometry"/>

	<arg name="NRD" value="CICPCriteriaNRD" />
	<arg name="ERD" value="CICPCriteriaERD" />
	<arg name="GSO" value="CLevMarqGSO" />

    <!-- Launch the demo bagfile -->
    <param name="/use_sim_time" value="true" type="bool" />
	<node pkg="rosbag" type="play" name="rosbag_player" args="--clock $(arg bag_file_full) $(arg bag_topic_remaps)" output="$(arg output)"/>

	<group ns="$(arg robot_ns)">
	    <!-- Run single-robot graphSLAM -->
	    <include file="$(find mrpt_graphslam_2d)/launch/graphslam.launch">
		    <arg name="output" value="$(arg output)" />
            <arg name="disable_MRPT_visuals" value="$(arg disable_MRPT_visuals)" />
  		    <arg name="verbosity" value="$(arg verbosity)" />
		    <arg name="start_rviz" value="$(arg start_rviz)" />
		    <arg name="rviz_file" value="$(arg rviz_file)" />
            <arg name="config_file" value="$(arg config_file)"  />
		    <arg name="anchor_frame_ID" value="$(arg anchor_frame_ID)" />
		    <arg name="base_link_frame_ID" value="$(arg base_link_frame_ID)" />
            <arg name="NRD" value="$(arg NRD)" />
            <arg name="ERD" value="$(arg ERD)" />
		    <arg name="GSO" value="$(arg GSO)" />
	    </include>
    </group>

	<node pkg="tf2_ros" type="static_transform_publisher" name="map_to_laser"
		args="0 0 0.02 0 0 0 $(arg base_link_frame_ID) $(arg laser_frame_ID)" />

</launch>
